[Storage] add new stroage account kind "FileStorage" and "BlockBlobStorage" and new feature fileaadIntegration

blueww opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

Rest Spec: https://microsoft.sharepoint.com/teams/AzureStorage/_layouts/15/Doc.aspx?sourcedoc=%7B00d45481-7f25-4170-a95e-82dbc6a3041d%7D&action=default&slrid=0d385f9e-a0fa-0000-758b-e5dde88dbd09

Pass review on private repo: 
New Kind:  https://github.com/Azure/azure-rest-api-specs-pr/pull/538
FileAAdIntegration: https://github.com/Azure/azure-rest-api-specs-pr/pull/508

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

blueww committed:
[Storage] Add api version 2018-07-01

blueww committed:
[Storage] update api version 2018-07-01

blueww committed:
[Storage] Add new Storage Account Kind and Sku

blueww committed:
[Storage] Remove the sku Premium_ZRS change as it won't be release in Ignite

blueww committed:
[Storage] Support FileAadIntegration

AutorestCI commented:
# Automation for azure-sdk-for-python

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-ruby

lmazuel committed:
Storage Python conf

lmazuel commented on specification/storage/resource-manager/Microsoft.Storage/stable/2018-07-01/storage.json:
```diff
@@ -0,0 +1,1998 @@
+        "tags": [
+          "Usage"
+        ],
+        "operationId": "Usage_List",
```
Why this renaming from "Usages" to "Usage"? That's a SDK breaking change.

lmazuel replied:
Ok, just see that it was "Usage", and then it was renamed in 2018-03-01-preview as "Usages", and now rename back to "Usage". So I guess it's ok

blueww replied:
it should be usages. I will fix it.

lmazuel commented:
Hi @blueww 
What is the status of 2018-03-01-preview? I see that management policies methods are not here in this PR. Does it mean it's missing here (and PR needs to be updated), or that management policies should be still called from 2018-03-01-preview? If the later is the truth, this means there is some Readme configuration to do to mix-up 2018-03-01-preview/2018-07-01 for SDK release.

AutorestCI commented:
# Automation for azure-sdk-for-go

AutorestCI commented:
# Automation for azure-sdk-for-java

AutorestCI commented:
# Automation for azure-sdk-for-node

lmazuel commented on specification/storage/resource-manager/Microsoft.Storage/stable/2018-07-01/storage.json:
```diff
@@ -0,0 +1,1998 @@
+        },
+        "isHnsEnabled": {
+          "type": "boolean",
+          "default": false,
```
Based on a recent issue found by @williexu in the CLI, please also remove this default value.

lmazuel commented on specification/storage/resource-manager/Microsoft.Storage/stable/2018-07-01/storage.json:
```diff
@@ -0,0 +1,1998 @@
+        },
+        "supportsHttpsTrafficOnly": {
+          "type": "boolean",
+          "default": false,
```
Based on a recent issue found by @williexu in the CLI, please also remove this default value.

blueww commented:
@lmazuel 
Management policies is not supported on 2018-07-01, it's only on 2018-03-01-preview now, so there will be a breaking change.
I have discussed with Management policies feature PM and SRP team, and get agreement not support Management policies  on 2018-07-01 and the SRP SDK based on it.

We will add back Management policies to a stable api version when the feature GA in the future.

blueww committed:
[Storage] modify per comments

blueww committed:
Merge branch 'newkindFileaad' of https://github.com/blueww/azure-rest-api-specs into newkindFileaad

blueww committed:
[Storage] Add the parameter location changes in 2018-03-01-preview to 2018-07-01

blueww committed:
[Storage] Add back Management Policy from 2017-03-01-preview

blueww committed:
[Storage] remove global usage as server change

blueww commented:
@lmazuel
I have added the Management policy feature back. And remove Global Usage list since not supported in new api version (as location usage added).
Please help to review.

lmazuel commented on specification/storage/resource-manager/Microsoft.Storage/stable/2018-07-01/storage.json:
```diff
@@ -0,0 +1,1921 @@
+        "tags": [
+          "LocationUsage"
+        ],
+        "operationId": "Usages_ListByLocation",
```
I feel we should keep “Usage” in 2018-07-01 and not “Usages”. It was like that for the 6 APIs between 2015-06-15 and 2018-02-01, and was changed in 2018-03-01-preview. I feel like we should keep the naming that all stable API were using

blueww replied:
@lmazuel 
The change is required by @anuchandy in review https://github.com/Azure/azure-rest-api-specs/pull/3014, since Using non-plural name for operation group is against SDK guideline . 
And finally we fix it in a new API version 2018-03-01-preview in https://github.com/Azure/azure-rest-api-specs/pull/3137, since this is a breaking change.
So I don't think we should keep using "Usages", instead of change it back.

lmazuel replied:
Fair enough :)

lmazuel commented on specification/storage/resource-manager/readme.md:
```diff
@@ -26,7 +26,32 @@ These are the global settings for the Storage API.
+input-file:
+- Microsoft.Storage/stable/2018-07-01/storage.json
+- Microsoft.Storage/stable/2018-07-01/blob.json
+- Microsoft.Storage/stable/2018-07-01/managementpolicy.json
```
Please put the new split file in “2018-03-01-preview” folder, and just reference it from here:
- Microsoft.Storage/stable/2018-07-01/storage.json
- Microsoft.Storage/stable/2018-07-01/blob.json
- Microsoft.Storage/stable/2018-03-01-preview/managementpolicy.json

blueww replied:
Fixed.

blueww committed:
[Storage] Add new Sku Premium_ZRS

blueww committed:
[Storage] move Management policy swagger to 2018-03-01-preview folder

blueww commented:
@lmazuel 
I have fixed the management policy swagger  location. And for Usage -> Usages, it's required by OneSDK team since "Using non-plural name for operation group is against SDK guideline ", I think we should keep using "Usages".
I also added new Sku "Premium_ZRS" per feature team requirement.

Since the release is target Ignite, and we need to raise and review SDK PR after the swagger PR is in, would you please help to review and merge the PR asap if no other issues.

Thanks very much!

hassanbabaie commented:
@blueww  Hi Wei, just a quick question I see this AAD support coming to AzureFiles, is this the same as what was just released at ignite as that  also required the use of Azure Active Directory Domain Services which is a real pain compared to say the new AAD for object storage which doesn't

Please let me know your thoughts

Hass

blueww commented:
@hassanbabaie 
You can find more details for the feature in https://azure.microsoft.com/en-us/blog/azure-active-directory-integration-for-smb-access-now-in-public-preview/
If you have any question or suggestions for this feature, you can contact AzureFiles@microsoft.com
